src/ray/raylet/scheduling_queue.cc

```diff
@@ -6,7 +6,8 @@ namespace {
 
 // Helper function to remove tasks in the given set of task_ids from a
 // queue, and append them to the given vector removed_tasks.
-void RemoveTasksFromQueue(ray::raylet::SchedulingQueue::TaskQueue &queue,
+template<typename TaskQueue>
+void RemoveTasksFromQueue(TaskQueue &queue,
```

Comment:
Dumb question, but why is this necessary?

Reply:
templating or the method itself? The method itself, IMHO, should be made a method of TaskQueue: `TaskQueue::RemoveTasks` to supplement `TaskQueue::RemoveTask`. Templating here is necessary, because the child class ReadyQueue is not a polymorphic type. So calling any method that takes `TaskQueue` as an argument will result in calling the parent class's implementation. 
Finally, why not use virtual? Great question! There's concern about performance implications of using virtual functions. The current PR design will make it straightforward to go from vanilla subclassing to polymorphic subclassing.